libs/app-theme/ui/src/lib/file-preview/file-preview.tsx

```diff
@@ -0,0 +1,329 @@
+/**
+ * Copyright (c) Mudita sp. z o.o. All rights reserved.
+ * For licensing, see https://github.com/mudita/mudita-center/blob/master/LICENSE.md
+ */
+
+import { FunctionComponent, useCallback, useEffect } from "react"
+import { Modal } from "../modal/modal"
+import styled, { css } from "styled-components"
+import { Typography } from "../typography/typography"
+import { IconSize, IconType, ModalLayer } from "app-theme/models"
+import { Icon } from "../icon/icon"
+import { ImagePreview } from "./image-preview"
+import { AnimatePresence, motion } from "motion/react"
+import { FilePreviewError, FilePreviewErrorCode } from "./file-preview-error"
+
+export { FilePreviewErrorCode } from "./file-preview-error"
+
+export interface FilePreviewFile {
+  id: string
+  name: string
+  type: string
+  path: string
+  extension: string
+  dataUrl?: string
+}
+
+interface Props {
+  opened?: boolean
+  file?: FilePreviewFile
+  filesCount: number
+  loading?: boolean
+  error?: FilePreviewErrorCode
+  onClose: VoidFunction
+  onNextFile?: VoidFunction
+  onPreviousFile?: VoidFunction
+  onDownload?: VoidFunction
+  onDelete?: VoidFunction
+  onRetry?: VoidFunction
+}
+
+export const FilePreview: FunctionComponent<Props> = ({
+  opened,
+  file,
+  filesCount,
+  loading,
+  error,
+  onClose,
+  onNextFile,
+  onPreviousFile,
+  onDownload,
+  onDelete,
+  onRetry,
+}) => {
+  const handleKeyDown = useCallback(
+    (event: KeyboardEvent) => {
+      switch (event.key) {
+        case "ArrowLeft":
+          event.preventDefault()
+          onPreviousFile?.()
+          break
+        case "ArrowRight":
+          event.preventDefault()
+          onNextFile?.()
+          break
+        case "D":
+        case "d":
+          event.preventDefault()
+          onDownload?.()
+          break
+        case "Delete":
+        case "Backspace":
+          event.preventDefault()
+          onDelete?.()
+          break
+        case "Escape":
+          event.preventDefault()
+          onClose()
+          break
+        default:
+          return
+      }
+    },
+    [onClose, onDelete, onDownload, onNextFile, onPreviousFile]
+  )
+
+  useEffect(() => {
+    if (opened) {
+      document.addEventListener("keydown", handleKeyDown)
+    } else {
+      document.removeEventListener("keydown", handleKeyDown)
+    }
+    return () => {
+      document.removeEventListener("keydown", handleKeyDown)
+    }
+  }, [handleKeyDown, opened])
+
+  return (
+    <Modal
+      opened={opened}
+      size={96}
+      customStyles={{ maxHeight: "64rem" }}
+      layer={ModalLayer.Default - 1}
+    >
+      <ModalContent>
+        <Header>
+          <Typography.P1 id={"file-preview-name"} color={"white"}>
+            [{file?.id}] {file?.name}
+          </Typography.P1>
+          <ActionButton onClick={onClose} aria-label="Close file preview">
```

Comment:
Aria labels here are hard-coded in English, should we translations for them as well? 🤷‍♂️